Don't crash on CIM classes with no superclass

edyoung opened the pull request:
## PR Summary

Fix #982 by checking for null on superclass property.

## PR Checklist

Note: Tick the boxes below that apply to this pull request by putting an `x` between the square brackets. Please mark anything not applicable to this PR `NA`.

- [x] PR has a meaningful title
    - [x] Use the present tense and imperative mood when describing your changes
- [x] Summarized changes
- [ ] User facing documentation needed
- [x] Change is not breaking
- [ ] Make sure you've added a new test if existing tests do not effectively test the code changed
- [x] This PR is ready to merge and is not work in progress
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready

edyoung commented:
Bergmeister's PR https://github.com/PowerShell/PSScriptAnalyzer/pull/985 addresses same issue, I am agnostic about which one to take. Would be nice to fix though...

bergmeister commented:
One PR build got stuck, which is an AppVeyor problem, therefore I triggered another build.
My PR is still WIP because I also wanted to remove the empty catch block, which then caused test failures, therefore, I think it is OK to take your PR with the minimal changes for fixing the NullReferenceException. I think we should add a test case for the bug that this PR fixes before merging the PR despite the fix being trivial.

bergmeister commented:
I think the error in the AppVeyor WMF4 build could actually due to the recent breaking change with TLS 1.2 here: https://github.com/dotnet/announcements/issues/77

edyoung commented:
OK, added a test case.

edyoung commented:
@JamesWTruher @bergmeister can we merge this now?

bergmeister commented:
Sorry, I will review this the next days (I want to test that that the added test would actually fail) but it looks OK otherwise.

edyoung commented:
Added -ErrorAction Stop - the additional try/catch/should check isn't needed as far as I can see - Pester fails the test because a terminating error occurs.

Changed the test data to be inline, but TBH I don't really think it's an improvement.

edyoung commented:
@JamesWTruher wondered if [path.io]::combine("testdrive:\", ...) would do the right thing on linux. I tried it, and I believe it does, but changed to Join-Path anyway. Also, turns out this rule basically doesn';t work on linux anyway because DscClassCache.Initialize() throws and the rest of the rule doesn't do anything. So I marked the test to be skipped on linux and mac like the others in this file.

edyoung commented:
@JamesWTruher if you're ok with this please go ahead & merge (FYI the main reason I care about this is that PowerShell Gallery script scanning process hits this exception fairly frequently and I want to cut the noise).

bergmeister commented:
@edyoung Since you have addressed @JamesWTruher and my comment about adding a test, I think it is OK to merge it in since the implementation has already been approved.
